### PR TITLE
feat: Correct and document `crimea.json`

### LINKED ISSUE #594: Unclear provenance of crimea.json dataset ('Nightingale's Rose')
The information carried by Florence Nightingale's [Rose Diagram](https://nrs.lib.harvard.edu/urn-3:hms.count:1177146?n=25) isn't just a dataset - it's a pivotal [moment](https://www.historyofinformation.com/detail.php?entryid=3815) in the history of data visualization and public health. Given the historical nature of this dataset, it's important that we represent it well. I've noticed some discrepancies that I believe we should address:

## Current Situation
1. Our dataset doesn't seem to match Nightingale's original published data. Nor does it match a [Protovis](https://mbostock.github.io/protovis/) [example](https://mbostock.github.io/protovis/ex/crimea-rose.html) by Mike Bostock. This makes it challenging to update `SOURCES.md`.
2. Bostock's visualization methods to reproduce the chart have been critiqued as not accurately reflecting Nightingale's original technique. This creates an opportunity to properly construct an example using vega or vega-lite.

## Details
The dataset in `crimea.json` appears to be derived from or inspired by a famous polar area [diagram](https://iiif.lib.harvard.edu/manifests/view/drs:7420433$23i) from Florence Nightingale's ["A contribution to the sanitary history of the British army during the late war with Russia"](https://nrs.lib.harvard.edu/urn-3:hms.count:1177146), which was later featured in this [Protovis example](https://mbostock.github.io/protovis/ex/crimea-rose.html) by Mike Bostock. 

As [noted](https://github.com/kgryte/gists/tree/master/d3/nightingales-rose) by @kgryte:

> Bostock's implementation, while visually similar to Nightingale's visualization, is wrong. First, the data is not correct. You can verify this in Nightingale's original work. Second, Bostock directly maps the wedge radius to deaths. This mistake is common. Instead, Nightingale represents deaths in terms of area, thus requiring the radius for each wedge to be calculated (for more information, see [Understanding Uncertainty's The Mathematics of Coxcombs](http://understandinguncertainty.org/node/214)). This discrepancy would be apparent if one displayed polar axes and allowed reading of radial values.

The [crimea.json](https://github.com/vega/vega-datasets/blob/main/data/crimea.json) hosted here does not match either [Bostock's dataset](https://mbostock.github.io/protovis/ex/crimea.js) or the Nightingale [table](https://nrs.lib.harvard.edu/urn-3:hms.count:1177146?n=21) that appears to sit behind her original polar area diagram. 

## Questions
1. Can we trace the exact provenance of our current `crimea.json` dataset?
2. Does `crimea.json` generate a similarly shaped diagram as the Bostock version? Perhaps it was just constructed to create a similar effect?
3. Should a Vega or Vega Lite example be created that addresses @kryte's critique?
4. Should a provisional description be added to the README.md file that notes this apparent discrepancy, or should we wait until this is resolved?
5. Should crimea.json eventually be modified to capture the actual data referenced in the original Nightingale source, for the sake of accuracy? It seems possible that at the time this dataset was uploaded, the full text of the Nightingale paper was difficult to access, while today the full text including the dataset is easily retrieved online.

### PR DESCRIPTION
**Pull request by dsmedia**
Resolves #594

# Tasks
* [X]  Replace `crimea.json` with version from [stdlib](https://github.com/vega/vega-datasets/blob/main/data/crimea.json)
* [X]  Add description, sources, and column descriptions for `crimea.json` to `_data/datapackage_additions.toml`

# Notes
* New json file includes an additional field for army_size, consistent with stdlib and Bostock version
* Dataset corrected to match the relevant table printed in [1859 Nightingale book](https://iiif.lib.harvard.edu/manifests/view/drs:7420433$21i)
* Row count and date values unchanged
* JSON formatted to match original `crimea.json` dataset

### CODE REVIEW

**Comment by dsmedia** [comment]
# Notes on Mortality Rate Data and Existing Visualizations

## 1. Mortality Rate Data Derivation
The mortality rate columns from Nightingale's [1859 publication](https://iiif.lib.harvard.edu/manifests/view/drs:7420433$21i) can be calculated from the raw death counts, making their inclusion in `crimea.json` redundant. The formula used is:

```
Annual Mortality Rate = (Deaths × 1000 × 12) / Army Size
```

### Verification
Here's a Python implementation that reproduces the original mortality rates:

```python
import pandas as pd

def transform_mortality_data(deaths_df):
    deaths_df['Diseases'] = (deaths_df['disease'] * 1000 * 12) / deaths_df['army_size']
    deaths_df['Wounds'] = (deaths_df['wounds'] * 1000 * 12) / deaths_df['army_size']
    deaths_df['Other'] = (deaths_df['other'] * 1000 * 12) / deaths_df['army_size']
    deaths_df['Month'] = pd.to_datetime(deaths_df['date']).dt.strftime('%B')
    deaths_df['Year'] = pd.to_datetime(deaths_df['date']).dt.year.astype('int64')
    return deaths_df[['Month', 'Year', 'Diseases', 'Wounds', 'Other']].round(1)
```

Sample output matches the original data:
```
April 1854: Diseases=1.4, Wounds=0.0, Other=7.0
December 1855: Diseases=25.3, Wounds=5.0, Other=7.8
March 1856: Diseases=3.9, Wounds=0.0, Other=9.1
```

## 2. Related Visualization Work
There's an existing Vega implementation by @avatorl that uses the transformed mortality rate data. See a [2022 blog post](https://www.powerofbi.org/2022/12/04/florence-nightingales-rose-diagram/) and [repository](https://github.com/avatorl/DataViz-Vega/tree/main/nightingale-rose) for details. This may be a good candidate for a Vega example @domoritz mentioned in [#594](https://github.com/vega/vega-datasets/issues/594#issuecomment-2546089689).

**Comment by dsmedia** [comment]
@dangotbanned Please note the diff shows very slight modifications to data filesizes in `datapackage.json` after I ran `build_datapackage.py`. If I can fix this (with a local configuration setting?), please let me know.

**Comment by dsmedia** [comment]
@dangotbanned also please be aware that when I attempted to break up the line length in the TOML multi-line strings below (for field description and source title) to keep to 80-100 characters per line, the resulting markdown table generated by `build-datapackage.py` was formatted incorrectly. As a result I kept it as one large line. Not sure if this is intended?


```toml
[[resources.schema.fields]]
name        = "wounds"
description = """Deaths from "Wounds and Injuries" which comprised: Luxatio (dislocation), Sub-Luxatio (partial dislocation), Vulnus Sclopitorum (gunshot wounds), Vulnus Incisum (incised wounds), Contusio (bruising), Fractura (fractures), Ambustio (burns) and Concussio-Cerebri (brain concussion)"""
```

```toml
[[resources.sources]]
title = """
Nightingale, Florence. A contribution to the sanitary history of the British army during the late war with Russia. London : John W. Parker and Son, 1859. Table II. Table showing the Estimated Average Monthly Strength of the Army; and the Deaths and Annual Rate of Mortality per 1,000 in each month, from April 1854, to March 1856 (inclusive), in the Hospitals of the Army in the East
"""
```

**Review by domoritz — APPROVED** [review]

**Comment by dangotbanned** [comment]
> @dangotbanned Please note the diff shows very slight modifications to data filesizes in `datapackage.json` after I ran `build_datapackage.py`. If I can fix this (with a local configuration setting?), please let me know.

Interesting 🤔 

My first thought would be maybe [os.stat_result.st_size](https://docs.python.org/3/library/os.html#os.stat_result.st_size) differs across platforms? 
Since the sizes you have are all smaller, I'm going to guess:

```py
import sys

sys.platform in {"darwin", "posix"} # @dsmedia
sys.platform == "win32"             # @dangotbanned
```

**Comment by dangotbanned** [comment]
https://github.com/vega/vega-datasets/pull/648#issuecomment-2557983063

@dsmedia I'll take a look at this today.

For this one, it could be the leading `"""\n` before the text actually starts?
```toml
[[resources.sources]]
title = """
Nightingale, Florence. A contribution to the sanitary history of the British army during the late war with Russia. London : John W. Parker and Son, 1859. Table II. Table showing the Estimated Average Monthly Strength of the Army; and the Deaths and Annual Rate of Mortality per 1,000 in each month, from April 1854, to March 1856 (inclusive), in the Hospitals of the Army in the East
"""
```

File: _data/datapackage_additions.toml [hunk 1]
```diff
@@ -205,6 +205,51 @@ path  = "https://www.gapminder.org/free-material/"
...
+
+Nightingale credits Dr. William Farr for compiling the data from the 1858 [Medical and Surgical 
+History of the British Army](http://resource.nlm.nih.gov/62510370R). The dataset categorizes 
+deaths into "zymotic" diseases (preventable infectious diseases), wounds/injuries, and other causes. 
+Covering the period from April 1854 to March 1856, the dataset includes monthly army strength 
+alongside mortality figures. Nightingale transformed this data into her now-famous [polar area 
+diagrams](https://iiif.lib.harvard.edu/manifests/view/drs:7420433$25i). 
+
+The annual mortality rates plotted in the chart can be calculated from the dataset using the formula 
+> (Deaths × 1000 × 12) / Army Size.
```
Comment (dangotbanned) [inline _data/datapackage_additions.toml:220]:
```suggestion
> (Deaths &times; 1000 &times; 12) &divide; Army Size. 
```

This was triggering an *ambiguous character* warning for me:

<details><summary>Warning</summary>
<p>


image

</p>
</details> 

<details><summary>Setting</summary>
<p>

https://code.visualstudio.com/docs/getstarted/settings#_default-settings

```jsonc

  // Controls whether characters are highlighted that can be confused with basic ASCII characters, except those that are common in the current user locale.
  "editor.unicodeHighlight.ambiguousCharacters": true,
```

</p>
</details> 


Switching to [HTML entities](https://www.w3schools.com/charsets/ref_html_entities_4.asp) seems to render the same - but is more explicit in your intent

- https://en.wikipedia.org/wiki/Multiplication_sign
- https://en.wikipedia.org/wiki/Division_sign

image

**Review by dangotbanned — APPROVED** [review]
Thanks for working on this @dsmedia!

**Pull request merged**